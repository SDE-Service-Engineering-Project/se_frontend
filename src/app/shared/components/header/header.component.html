<nav class="navbar navbar-expand-lg bg-primary" data-bs-theme="dark">
  <div class="container-fluid">
    <a class="navbar-brand" routerLink="/home">Home</a>
<button class="navbar-toggler" type="button" data-testid="toggle-btn" (click)="isNavbarCollapsed = !isNavbarCollapsed" data-bs-toggle="collapse" data-bs-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle navigation">
  <span class="navbar-toggler-icon"></span>
</button>
    <div [ngbCollapse]="isNavbarCollapsed" class="collapse navbar-collapse" id="navbarSupportedContent">
      <ul class="navbar-nav me-auto mb-2 mb-lg-0">
        <li class="nav-item" *ngFor="let item of navItems">
          <a class="nav-link" [attr.data-testid]="item.testId" routerLinkActive="active-link"
             [routerLink]="item.path">{{item.title}}</a>
        </li>
      </ul>
      <ul class="navbar-nav">
        <li class="nav-item">
          <a id="logout" class="nav-link" data-testid="nav-logout" (click)="logout()" >Logout</a>
        </li>
      </ul>
    </div>
  </div>
</nav>
