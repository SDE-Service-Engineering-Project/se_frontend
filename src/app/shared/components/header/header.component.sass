.navbar-brand
  color: white

.nav-link
  color: white

.navbar-nav
  display: flex

#logout
  float: right
  cursor: pointer
