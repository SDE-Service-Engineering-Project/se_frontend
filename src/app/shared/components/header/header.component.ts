import { Component } from '@angular/core';
import { Router } from '@angular/router';
import { StorageService } from '../../../services/storage.service';

@Component({
  selector: 'app-header',
  templateUrl: './header.component.html',
  styleUrls: ['./header.component.sass'],
})
export class HeaderComponent {
  isNavbarCollapsed = false;
  navItems = [
    {
      title: 'Catalog',
      path: '/catalog',
      testId: 'nav-item-catalog',
    },
    {
      title: 'Rented Cars',
      path: '/cars',
      testId: 'nav-item-cars',
    },
    {
      title: 'Find a Location',
      path: '/maps',
      testId: 'nav-item-maps',
    },
  ];

  constructor(private storageService: StorageService, private router: Router) {}

  logout(): void {
    this.storageService.removeAll();
    this.router.navigate(['/']);
  }
}
