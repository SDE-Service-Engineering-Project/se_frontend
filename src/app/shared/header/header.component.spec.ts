import { ComponentFixture, TestBed } from '@angular/core/testing';

import { HeaderComponent } from './header.component';
import { createComponentFactory, Spectator } from '@ngneat/spectator/jest';

describe('HeaderComponent', () => {
  let component: HeaderComponent;
  let fixture: ComponentFixture<HeaderComponent>;
  let spectator: Spectator<HeaderComponent>;

  const createComponent = createComponentFactory({
    component: HeaderComponent,
  });

  beforeEach(() => {
    spectator = createComponent();
    fixture = spectator.fixture;
    component = spectator.component;
  });

  it('should render the component', () => {
    expect(fixture.debugElement.nativeElement).toMatchSnapshot();
  });

  it('should navigate to', function () {});
});
