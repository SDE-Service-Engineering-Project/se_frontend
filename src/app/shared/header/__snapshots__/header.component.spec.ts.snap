// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderComponent should render the component 1`] = `
<div
  id="root0"
>
  <nav
    class="navbar navbar-expand-lg bg-body-tertiary"
  >
    <div
      class="container-fluid"
    >
      <a
        class="navbar-brand"
        href="#"
      >
        Navbar
      </a>
      <button
        aria-controls="navbarSupportedContent"
        aria-expanded="false"
        aria-label="Toggle navigation"
        class="navbar-toggler"
        data-bs-target="#navbarSupportedContent"
        data-bs-toggle="collapse"
        type="button"
      >
        <span
          class="navbar-toggler-icon"
        />
      </button>
      <div
        class="collapse navbar-collapse"
        id="navbarSupportedContent"
      >
        <ul
          class="navbar-nav me-auto mb-2 mb-lg-0"
        >
          <li
            class="nav-item"
          >
            <a
              class="nav-link"
              href="#"
            >
              Logout
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class="nav-link"
              href="#"
            >
              Catalog
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class="nav-link"
              href="#"
            >
              Rented Cars
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class="nav-link disabled"
            >
              Find a location
            </a>
          </li>
        </ul>
      </div>
    </div>
  </nav>
</div>
`;
